### PR TITLE
[hotfix] fn:serialize issues

### LINKED ISSUE #4702: [BUG] issue serializing a sequence of nodes
**Describe the bug**

When trying to serialize a sequence of nodes, only the first gets serialized properly and the following nodes get atomized.
`(<a>foo</a>, <b>bar</b>) => serialize()` produces `"<a>foo</a> bar"`  on current eXist 6.1.0. (Tested at http://exist-db.org/exist/apps/eXide/index.html)

**Expected behavior**

The output should be `"<a>foo</a><b>bar</b>"` 

**To Reproduce**

```Xquery
xquery version "3.1";

module namespace mytest="http://mytest";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare %test:assertEquals("<a>foo</a><b>bar</b>") function mytest:a() {
    (<a>foo</a>, <b>bar</b>) => serialize()
};

declare %test:assertEquals("<a>foo</a><b>bar</b>") function mytest:b() {
    document { <a>foo</a>, <b>bar</b> } => serialize()
};
```

**Context (please always complete the following information):**
 - OS:  macOS 12.6.2
 - eXist-db version: 6.1.0
 - Java 1.8.0_342

### LINKED ISSUE #4704: [BUG] item-separator cannot be set to empty string
**Describe the bug**

The serialisation option `item-separator` introduced in existdb-6.1.0 does not allow to set it to an empty string.
The default item-separator is used instead.

**Expected behavior**

The above to work.

**To Reproduce**

```Xquery
xquery version "3.1";

module namespace ser="http://exist-db.org/xquery/test";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare
    %test:assertEquals("aaabbb")
function ser:serialize-atomic-empty-separator() {
    serialize(("aaa", "bbb"),
        map {"item-separator": ""})
};

```

**Context (please always complete the following information):**
 - OS: macOS 12
 - eXist-db version: 6.1.0
 - Java Version Java8u352

**Additional context**
- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`? none

### LINKED ISSUE #4705: [BUG] serialise does not skip over empty text nodes
**Describe the bug**

The XQuery specification states that any element in a sequence that is normalised to an empty string must be skipped.
Given a sequence of text nodes $text-nodes declared as 

```
let $text-nodes := (<a>[</a>, <a> </a>, <a>]</a>)/text()
```

where the entry in the middle only contains whitespace, it is to be skipped and no extra item-separator be added to the serialisation.

```xquery
serialize($text-nodes, map {"item-separator": "|"})
```

With exist-6.1.0 this produces `[| |]`

**Expected behavior**

The above to evaluate to `[|]`

**To Reproduce**

```Xquery
xquery version "3.1";

module namespace ser="http://exist-db.org/xquery/test";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare
    %test:assertEquals("[|]")
function ser:sequence-skip-empty-text-node() {
    fn:serialize((<a>[</a>, <a>
    </a>, <a>]</a>)/text(), map {"item-separator": "|"})
};
```

**Context (please always complete the following information):**

- Build: eXist-6.1.0
- Java: 1.8.0_362 (Azul Systems, Inc.)
- OS: Mac OS X 12.6.2 (aarch64)

**Additional context**

- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`?

### LINKED ISSUE #4706: [BUG] fn:serialize returns empty string for arrays
**Describe the bug**

1. `serialize([1,2])` yields ''
2. `serialize([1,2], map{"item-separator": "|"})` yields ''

**Expected behavior**

1. '1 2'
2. '1|2'

**To Reproduce**

```Xquery
xquery version "3.1";

module namespace t="http://exist-db.org/xquery/test";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare
    %test:assertEquals("1 2")
function t:test() {
    serialize([1, 2])
};

declare
    %test:assertEquals("1|2")
function t:test-with-item-separator() {
    serialize([1, 2], map{"item-separator": "|"})
};
```

**Context (please always complete the following information):**
- Build: eXist-6.2.0-SNAPSHOT (7718b141f00baf3172c5115c0d96aff44ec8e591)
- Java: 1.8.0_362 (Azul Systems, Inc.)
- OS: Mac OS X 12.6.2 (aarch64)

**Additional context**

- How is eXist-db installed? built from source
- Any custom changes in e.g. `conf.xml`? none

### PR DESCRIPTION
**Pull request by line-o**
### Description:

- apply item separator _after_ string normalisation: fixes #4702
- allow empty item separator: fixes #4704
- skip empty values: fixes #4705
- item-separator applies to array members: fixes #4706 

Note:
The provided test cases by @peterstadler had an expected value of `"<a>foo</a><b>bar</b>"`
This had to be changed to `"<a>foo</a> <b>bar</b>"` because of the _changed_ default item separator. 

### Reference:

https://www.w3.org/TR/xslt-xquery-serialization-31/#serdm

### Type of tests:

XQSuite tests

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/functions/fn/FunSerialize.java [hunk 1]
```diff
@@ -144,32 +144,20 @@ public static Sequence normalize(final Expression callingExpr, final XQueryConte
         if (input.isEmpty())
             // "If the sequence that is input to serialization is empty, create a sequence S1 that consists of a zero-length string."
             {return StringValue.EMPTY_STRING;}
+        final String _separator = itemSeparator == null ? DEFAULT_ITEM_SEPARATOR : itemSeparator;
```
Comment (line-o) [inline exist-core/src/main/java/org/exist/xquery/functions/fn/FunSerialize.java:147]: We should get rid of this null check here.

**Comment by line-o** [comment]
I consider this PR done. If someone finds an edge case I did not cover please comment.

**Comment by joewiz** [comment]
Thanks, Juri!! I'm curious how the PR performs on XQTS's serialize tests, compared to before.

**Comment by line-o** [comment]
@joewiz Yes, I want to see that comparison as well :)
My expectation is that several additional tests now pass but I do not have time to take on that task.

**Comment by line-o** [comment]
@joewiz hmm, it even looks like the XQTS job is failing at the moment by looking at the output :(
https://github.com/eXist-db/exist/actions/runs/4024734250/jobs/6917098711

> Caused by: java.lang.NoSuchMethodError: org.apache.logging.slf4j.Log4jLoggerFactory.<init>(Lorg/apache/logging/slf4j/Log4jMarkerFactory;)V

> Warning: No files were found with the provided path: /tmp/xqts-output. No artifacts will be uploaded.

**Review by dizzzz — APPROVED** [review]

**Review by reinhapa — APPROVED** [review]
